### DISCUSSION TITLE
Error transforming Node to Any when coming from another Either

### DISCUSSION THREAD

**Discussion by gilcu2** [VirtusLab/scala-yaml#145] [Q&A] [answered]
Hi, here problem reproduction. Scala 3.1.1, "org.virtuslab" %% "scala-yaml" % "0.0.4",

```scala
import org.virtuslab.yaml.*

object Yamltest extends App {

  def toNode(s: String): Either[YamlError, Node] = s.asNode

  def toEitherString(e: Either[YamlError, Node]):Either[String,Node] = e match {
    case Left(x) => Left(x.msg)
    case Right(x) => Right(x)
  }

  def toAny(e: Either[String, Node]): Either[String, Any] =  e.flatMap(_.as[Any])


  val yaml = "1"
  val possible_node = toNode(yaml)
  val possible_node_string_either = toEitherString(possible_node)
  val possible_any = toAny(possible_node_string_either)

  println(possible_node)
  println(possible_any)
}
```

When compile get: no implicit argument of type org.virtuslab.yaml.LoadSettings was found for parameter settings of method as in trait Node
  def toAny(e: Either[String, Node]): Either[String, Any] =  e.flatMap(_.as[Any])

**Comment by kpodsiad** [comment, marked as answer]
You're flatmapping and expecting `Either[String, Any]` but `as[T]` returns `Either[YamlError, T]`. `e.flatMap(_.as[Any].left.map(_.toString))` should fix this error. 

I wonder if the compiler should have a better message.

**Reply by gilcu2** [reply]
Thanks very much. The message lost me

**Reply by kpodsiad** [reply]
Yes, that message is misleading a bit. I talked to someone on the compiler team and this might be turned into an issue.
